bokeh/util/serialization.py

```diff
@@ -7,55 +7,132 @@
 
 {binary_array_types}
 
-"""
+'''
 from __future__ import absolute_import
 
+import logging
+log = logging.getLogger(__name__)
+
 import base64
+import datetime as dt
 import math
+import time
 
 from six import iterkeys
 
-from bokeh.util.string import format_docstring
+import numpy as np
+
+from .string import format_docstring
 from .dependencies import import_optional
 
-is_numpy = None
-
-try:
-    import numpy as np
-    is_numpy = True
-    BINARY_ARRAY_TYPES = set([
-        np.dtype(np.float32),
-        np.dtype(np.float64),
-        np.dtype(np.uint8),
-        np.dtype(np.int8),
-        np.dtype(np.uint16),
-        np.dtype(np.int16),
-        np.dtype(np.uint32),
-        np.dtype(np.int32),
-    ])
-except ImportError:
-    is_numpy = False
-    BINARY_ARRAY_TYPES = set()
```

Comment:
NumPy was imported directly in `json_encoder.py` for some time which AFAIK meant Bokeh could not function in any way without numpy. I think we should give up the pretense that Bokeh can be used without numpy.